### PR TITLE
DRILL-6331: Added tests to verify that optimizations like filter push…

### PR DESCRIPTION
**Pull request by agozhiy**
… down work with native Hive reader.

### CODE REVIEW

File: framework/resources/Functional/hive/hive_optimizations/data/DRILL_6331_count_into_scan_pushdown_managed_table.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+set `store.hive.optimize_scan_with_native_readers` = true;
+--@test
```
Comment (Agirish) [inline framework/resources/Functional/hive/hive_optimizations/data/DRILL_6331_count_into_scan_pushdown_managed_table.sql]: Don't we need to reset the option after the test? 
Also, please remove the @test annotation - lets use that only when we have multiple SQL queries within a given test file and only one of them needs to be tested for correctness.

Comment (agozhiy) [reply framework/resources/Functional/hive/hive_optimizations/data/DRILL_6331_count_into_scan_pushdown_managed_table.sql]: Done.

**Comment by agozhiy** [comment]
@Agirish, addressed your comment, could you review?

**Comment by Agirish** [comment]
Thanks @agozhiy. I'll have it run through regression tests and have it merged.

**Comment by Agirish** [comment]
+1. LGTM. Tests passed.

**Pull request merged**